backend/tournesol/tests/test_api_comparison.py

```diff
@@ -923,3 +932,76 @@ def test_invalid_criteria_in_comparison(self):
         self.assertContains(
             response, "not a valid criteria", status_code=status.HTTP_400_BAD_REQUEST
         )
+
+
+class ComparisonWithMehestanTest(TestCase):
+    def setUp(self):
+        self.poll = PollFactory(algorithm=ALGORITHM_MEHESTAN)
+        CriteriaRankFactory(poll=self.poll, criteria__name="criteria1")
+        CriteriaRankFactory(poll=self.poll, criteria__name="criteria2", optional=True)
+
+        self.entities = VideoFactory.create_batch(3)
+        self.user1, self.user2 = UserFactory.create_batch(2)
+
+        comparison = ComparisonFactory(
+            poll=self.poll,
+            user=self.user1,
+            entity_1=self.entities[0],
+            entity_2=self.entities[1],
+        )
+
+        for (criteria, score) in [("criteria1", 1), ("criteria2", 2)]:
+            ComparisonCriteriaScoreFactory(
+                comparison=comparison,
+                criteria=criteria,
+                score=score,
+            )
+
+        self.client = APIClient()
+
+    def test_update_individual_scores_after_new_comparison(self):
+        call_command("ml_train")
+
+        self.assertEqual(ContributorRatingCriteriaScore.objects.count(), 4)
+        self.assertEqual(EntityCriteriaScore.objects.count(), 4)
```

Comment:
Should we also check the individual scores are not present before making the post?

```python
        self.assertEqual(
            ContributorRatingCriteriaScore.objects
                .filter(contributor_rating__user=self.user2)
                .count(),
            0
        )
```